src/plugins/saved_objects_management/public/lib/resolve_saved_objects.ts

```diff
@@ -280,7 +276,10 @@ export async function resolveSavedObjects(
   let importedObjectCount = 0;
   const failedImports: FailedImport[] = [];
   // Start with the index patterns since everything is dependent on them
-  await awaitEachItemInParallel(docTypes.indexPatterns, async (indexPatternDoc) => {
+  // As the confirmation opens a modal, and as we only allow one modal at a time
+  // (opening a new one close the previous with a rejection)
+  // we can't do that in parallel
+  for (const indexPatternDoc of docTypes.indexPatterns) {
```

Comment:
So, this was present for a very long time (since  core's `ModalService` presence), as we are dismissing any previous modal when opening a new one

https://github.com/elastic/kibana/blob/395d62124989811e32f4f9e8c27975f2b776559b/src/core/public/overlays/modal/modal_service.tsx#L123-L128

The `awaitEachItemInParallel` was causing multiple modals to eventually open at the same time, dismissing the previous one when the next one kicked in. 

Using a sort of confirmation queue here to pool the modal promises and only open the next one when the previous one is answered by the user is a nightmare, as some confirmations modal are opened as down as savedobject loaders underlying implementations in some cases. As this is only impacting the legacy imports, removing the parallel work and doing one object at a time was the only sane option.

From a quick manual benchmark, it doesn't even impact the process time that much.

Reply:
So the modern approach won't be affected by this fix?

Reply:
Yea. the modern approach is not impacted by this bug, and is not relying on this mechanism to prompt user confirmations, so the PR has no impact on it